sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSQLQueryPlanner.java

```diff
@@ -116,8 +116,43 @@ public QueryPlanner createPlanner(
     return modifyRuleSetsForZetaSql(BeamRuleSets.getRuleSets());
   }
 
-  /** Returns true if the argument contains any user-defined Java functions. */
-  static boolean hasUdfInProjects(RelOptRuleCall x) {
+  /**
+   * Returns true if the arguments only contain any user-defined Java functions, otherwise returns
```

Comment:
We should add a few more sentences explaining exactly what this means too. It's kind of confusing right now because not all instances of `SqlUserDefinedFunction` are actually user defined, but someone who reads this code might not realize that unless we clarify.

```suggestion
   * Returns true if the arguments only contain user-defined Java functions, otherwise returns
```

Reply:
Added to comment to clarify what is user-defined java functions.